### ISSUE TITLE
Question about the code

### ISSUE THREAD

**Issue by LiJiahang617** [HVision-NKU/MaskCLIPpp#9]
Thanks for your impressive work! When I read your code, I found the following part that I can not understand:
1. In your config file, why you use panoptic settings to eval your model instead of semantic seg? Just like below:

![Image](https://github.com/user-attachments/assets/e099c28d-1987-4e5c-bd60-5f81748faf58)
I found your code does contain the code about semantic seg setting for ADE20K-150:

![Image](https://github.com/user-attachments/assets/63b31a52-cfc5-435b-b5ae-e63cafe31092)

2. What is the difference between the name of visual_encoder_f and visual_encoder? Is it mean visual_encoder(finetune)? I do not understand why you implement two identical class in one forward func?

It would be very helpful if you could tell me these details! Thanks.

**Comment by ashun989** [comment]
1. As you can see, there are two types of data annotations in ADE20K(150): semantic annotation or panoptic annotation. Based on our observation, there are a small number of ignore regions (label=255, visualized as fully black) in the panoptic annotations, while such regions do not exist in the semantic annotation. This difference causes an extremely minor variation in the mIoU performance on ADE20K. Our final choice of using the panoptic annotation of ADE20K(150) also follows the settings of prior works, such as [FC-CLIP](https://github.com/bytedance/fc-clip/blob/2b0bbe213070d44da9182530fa2e826fef03f974/configs/coco/panoptic-segmentation/fcclip/fcclip_convnext_large_eval_ade20k.yaml#L28) and [MAFT+](https://github.com/jiaosiyu1999/MAFT-Plus/blob/fd12806df651d309883229de9503e40533f92689/configs/semantic/eval.yaml#L15).

2. As shown in the architecture diagram, MaskCLIP++ does not share the backbone with the mask generator. The backbone of MaskCLIP++ is named `visual_encoder`, which receives fine-tuning during training. In contrast, the backbone of the mask generator is named `visual_encoder_f` and is only used during OVS inference.

**Comment by LiJiahang617** [comment]
Thanks a lot, I understand now. Best wish!
